msgraph/claims_mapping_policy.go

```diff
@@ -0,0 +1,169 @@
+package msgraph
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/manicminer/hamilton/odata"
+)
+
+type ClaimsMappingPolicyClient struct {
+	BaseClient Client
+}
+
+// NewClaimsMappingPolicyClient returns a new ClaimsMappingPolicyClient
+func NewClaimsMappingPolicyClient(tenantId string) *ClaimsMappingPolicyClient {
+	return &ClaimsMappingPolicyClient{
+		BaseClient: NewClient(VersionBeta, tenantId),
+	}
+}
+
+// Create creates a new ClaimsMappingPolicy.
+func (c *ClaimsMappingPolicyClient) Create(ctx context.Context, policy ClaimsMappingPolicy) (*ClaimsMappingPolicy, int, error) {
+	var status int
+
+	body, err := json.Marshal(policy)
+	if err != nil {
+		return nil, status, fmt.Errorf("json.Marshal(): %v", err)
+	}
+
+	resp, status, _, err := c.BaseClient.Post(ctx, PostHttpRequestInput{
+		Body:             body,
+		ValidStatusCodes: []int{http.StatusCreated},
```

Comment:
You can request the full metadata in a POST request, and the response, whilst larger, will contain the all important `@odata.id`. For example:

```suggestion
		OData:            odata.Query{Metadata: odata.MetadataFull},
		ValidStatusCodes: []int{http.StatusCreated},
```

Alas the `@odata.id` returned by the API can arrive in a couple different formats, one of which paradoxically isn't accepted by the API - so the `odata.Id` type has its own workaround for this in its [`MarshalJSON()` method](https://github.com/manicminer/hamilton/blob/main/odata/odata.go#L17-L40). Whilst testing this I noticed the regex doesn't work for nested object URIs, so I've opened a PR #152 to fix that, which you'll need.

Reply:
We rebased on the bug fix branch to pull that change in. Our tests locally are passing. Thank you.